src/controllers/StudentController.ts

```diff
@@ -2,12 +2,97 @@ import { Student } from "../entities/Student";
 
 import { Request, Response } from "express";
 import { Repository } from "typeorm";
+import { validate } from "class-validator";
 
 export const StudentController = (studentRepository: Repository<Student>) => {
   const getAll = async (req: Request, res: Response): Promise<Response> => {
-    const students = await studentRepository.find();
-    return res.status(200).json(students);
+    try {
+      const students = await studentRepository.find();
+      return res.status(200).json(students);
+    } catch (error) {
+      return res.status(500).json({ error: "Unexpected DB error" });
+    }
   };
 
-  return { getAll };
+  const getById = async (req: Request, res: Response): Promise<Response> => {
+    const studentId = req.params.studentId;
+
+    try {
+      const student = await studentRepository.findOne(studentId);
+
+      if (student) {
+        return res.status(200).json(student);
+      }
+      return res.status(404).json({ error: "Student not found" });
```

Comment:
Prioriza responder de último la respuesta correcta, y de último los casos de error.
```suggestion
      if (!student) {
        return res.status(404).json({ error: "Student not found" });
      }
      return res.status(200).json(student);
```

Reply:
Me gusta la idea de tratar a los ifs como guards, donde se entre en caso que algo salga fuera del happy path. Ya hice el cambio